pyvisfile/xdmf/__init__.py

```diff
@@ -0,0 +1,1119 @@
+        return DataItemNumberType.UInt
+    elif dtype.type in (np.float16, np.float32, np.float64, np.float128):
+        return DataItemNumberType.Float
+    else:
+        raise ValueError(f"unsupported dtype: '{dtype}'")
+
+
+def _data_item_format_from_str(text):
+    if not isinstance(text, str):
+        return DataItemFormat.XML
+
+    # NOTE: this handles two cases (not meant to be too smart about it)
+    # 1. if the text is just `some_file.bin` or `some_file.out`, we assume it's
+    #    a custom binary file.
+    # 2. if the text is `some_file.h5:/group/dataset`, we assume it's an HDF5
+    #    file.
+
+    if ":" not in text:
+        return DataItemFormat.Binary
+
+    try:
+        filename, _ = text.split(":")
+    except ValueError as exc:
+        raise ValueError("cannot determine format from text") from exc
+
+    if filename.endswith(".h5"):
+        return DataItemFormat.HDF
+    else:
+        raise ValueError("cannot determine format from text")
+
+
+def _system_to_xdmf_endian():
+    import sys
+    if sys.byteorder == "little":
+        endian = DataItemEndian.Little
+    elif sys.byteorder == "big":
+        endian = DataItemEndian.Big
+    else:
+        endian = DataItemEndian.Native
+
+    return endian
+
+
+class DataItemType(enum.Enum):
+    """Data layout of an item."""
+    Uniform = enum.auto()
+    HyperSlab = enum.auto()
+    Function = enum.auto()
+
+
+class DataItemNumberType(enum.Enum):
+    """Basic number types for an item."""
+    Char = enum.auto()
+    UChar = enum.auto()
+    Int = enum.auto()
+    UInt = enum.auto()
+    Float = enum.auto()
+
+
+class DataItemFormat(enum.Enum):
+    """Format in which the item is stored."""
+    XML = enum.auto()
+    HDF = enum.auto()
+    Binary = enum.auto()
+    # NOTE: unsupported as it's pretty much undocumented
+    TIFF = enum.auto()
+
+
+class DataItemEndian(enum.Enum):
+    """Endianess of the data stored in the item."""
+    Native = enum.auto()
+    Big = enum.auto()
+    Little = enum.auto()
+
+
+class DataItem(XdmfElement):
+    """
+    .. attribute:: dimensions
+
+    .. automethod:: __init__
+    """
+
+    def __init__(self, *,
+            dimensions: Optional[Tuple[int, ...]] = None,
+            name: Optional[str] = None,
+            itype: Optional[DataItemType] = DataItemType.Uniform,
+            ntype: Optional[DataItemNumberType] = DataItemNumberType.Float,
+            precision: Optional[int] = 4,
+            reference: Optional[str] = None,
+            function: Optional[str] = None,
+            endian: Optional[DataItemEndian] = DataItemEndian.Native,
+            dformat: Optional[DataItemFormat] = DataItemFormat.XML,
+            parent: Optional[Element] = None,
+            data: Optional[str] = None,
+            ):
+        """
+        :param parent: if provided, *self* is appended to the element.
+        :param reference: name of another :class:`DataItem`.
```

Comment:
What does this do?

Reply:
The reference? Mostly meant to denote a `DataItem` which is just a link to some other `DataItem` somewhere, like in [this example](https://www.xdmf.org/index.php/XDMF_Model_and_Format#SubSet).